env/dev/clj/dev.clj

```diff
@@ -0,0 +1,63 @@
+(ns dev
+  "Userspace functions you can run by default in your local REPL."
+  (:require
+    [chocolate.config :refer [env]]
+    [clojure.pprint]
+    [clojure.spec.alpha :as s]
+    [expound.alpha :as expound]
+    [mount.core :as mount]
+    [conman.core :as conman]
+    [chocolate.figwheel :refer [start-fw stop-fw cljs]]
+    [luminus-migrations.core :as migrations]
+    [chocolate.core :refer [start-app]]
+    [chocolate.db.core]))
+
+
+(alter-var-root #'s/*explain-out* (constantly expound/printer))
+
+(add-tap (bound-fn* clojure.pprint/pprint))
+
+(defn start
+  "Starts application.
+  You'll usually want to run this on startup."
+  []
+  (mount/start-without #'chocolate.core/repl-server))
+
+(defn stop
+  "Stops application."
+  []
+  (mount/stop-except #'chocolate.core/repl-server))
+
+(defn restart
+  "Restarts application."
+  []
+  (stop)
+  (start))
+
+(defn restart-db
+  "Restarts database."
+  []
+  (mount/stop #'chocolate.db.core/*db*)
+  (mount/start #'chocolate.db.core/*db*)
+  (binding [*ns* 'chocolate.db.core]
+    (conman/bind-connection chocolate.db.core/*db* "sql/queries.sql")))
+
+(defn reset-db
+  "Resets database."
+  []
+  (migrations/migrate ["reset"] (select-keys env [:database-url])))
+
+(defn migrate
+  "Migrates database up for all outstanding migrations."
+  []
+  (migrations/migrate ["migrate"] (select-keys env [:database-url])))
+
+(defn rollback
+  "Rollback latest database migration."
+  []
+  (migrations/migrate ["rollback"] (select-keys env [:database-url])))
+
+(defn create-migration
+  "Create a new up and down migration file with a generated timestamp and `name`."
+  [name]
+  (migrations/create name (select-keys env [:database-url])))
```

Comment:
So a group of debugging aides all in one namespace?  Cool.
Does this only get compiled in a dev build?

Reply:
That is my understanding